src/Compiler.ts

```diff
@@ -42,9 +47,23 @@ function compile(fileNames: string[], options: ts.CompilerOptions): void {
         options.outDir = options.rootDir;
     }
 
+    if (!('addHeader' in options)) {
+        options.addHeader = true;
+    }
+
+    if (!options.luaLibDir) {
+        options.luaLibDir = options.outDir;
+    }
+    else {
+        options.luaLibDir = path.join(options.outDir, options.luaLibDir);
+    }
+    if (!fs.existsSync(options.luaLibDir)){
+        fs.mkdirSync(options.luaLibDir);
```

Comment:
Pretty sure mkdir sync is not recursive. In this case this will error on a nested path.

Reply:
Typescript has a function to recursively create files / directories. not sure If that function is exposed tho.